pulpcore/cli/common/generic.py

```diff
@@ -40,6 +40,34 @@ def format_help_text(
 
 
 class PulpOption(click.Option):
+    def __init__(
+        self,
+        *args: Any,
+        needs_plugin: Optional[str] = None,
+        min_version: Optional[str] = None,
+        max_version: Optional[str] = None,
+        **kwargs: Any,
+    ):
+        if (min_version or max_version) and not needs_plugin:
+            raise click.ClickException(
```

Comment:
I'd reserve ClickExceptions to user errors. This one is an implementation error. Should be ok to use assert here.